src/rdb.c

```diff
@@ -3260,7 +3261,10 @@ int rdbLoadRioWithLoadingCtx(rio *rdb, int rdbflags, rdbSaveInfo *rsi, rdbLoadin
         /* If there is no slot info, it means that it's either not cluster mode or we are trying to load legacy RDB file.
          * In this case we want to estimate number of keys per slot and resize accordingly. */
         if (should_expand_db) {
-            expandDb(db, db_size);
+            if (expandDb(db, db_size) == C_ERR) {
+                serverLog(LL_WARNING, "OOM in dict try expand");
+                return C_ERR;
+            }
```

Comment:
i'm not sure this graceful error handling is needed.
we do that when loading individual keys, in order to fail gracefully on a corrupt or malicious RESTORE command.
but i'm not sure that's needed on RDB loading / full-sync.
was there any discussion about that that i missed?

p.s. if we do that, we should also do it for the expires dict and the per-slot dicts.

Reply:
is this related to https://github.com/redis/redis/pull/11695#discussion_r1292757596?

p.s. i think the log message is insufficient, we don't know which dict we failed to allocate and what was it's size.
i suggest to either improve the log message, or pass a boolean argument to expandDB to cause it to use zcalloc instead of ztrycalloc in which case it'll panic with full stack trace and the requested size.

Reply:
Yeah, I was trying to address that particular comment.